types/firestore.d.ts

```diff
@@ -1895,6 +1916,34 @@ declare namespace FirebaseFirestore {
     valueOf(): string;
   }
 
+  /**
+   * Builds a Firestore data bundle with results from the given document and query snapshots.
+   */
+  export class BundleBuilder {
+    /**
+     * Adds a Firestore document snapshot or query snapshot to the bundle.
+     * Both the documents data and the query read time will be included in the bundle.
+     *
+     * @param {DocumentSnapshot=} documentSnapshot A document snapshot to add.
+     * @returns {BundleBuilder} This instance.
+     */
+    add(documentSnapshot: DocumentSnapshot): BundleBuilder;
```

Comment:
Should `DocumentSnapshot` (and `QuerySnapshot`) below be generic (for converter support)? Does the BundleBuilder respect the converters as it generates the Proto?

Reply:
All it does is call `toDocumentProto()` on the snapshot, I don't think it is necessary to make it generic?

Reply:
There is actually no such thing as a non-generic DocumentSnapshot in our API. The DocumentSnapshot type defaults to `DocumentSnapshot<DocumentData>`, which means that a custom type such as `DocumentSnapshot<Foo>` cannot be passed. This should be:

```
add<T>(documentSnapshot: DocumentSnapshot<T>): BundleBuilder;
```

Same for QuerySnapshot.

Reply:
Done, thanks.